### PR TITLE
bpo-41798: Allocate the expat_CAPI on the heap memory.

### PR DESCRIPTION
**Pull request by shihai1991**
- Issue: #85964

### CODE REVIEW

File: Modules/pyexpat.c [hunk 1]
```diff
@@ -1836,6 +1836,15 @@ add_features(PyObject *mod)
 }
 #endif
 
+static void
+pyexpat_destructor(PyObject *op)
+{
+    void *p = PyCapsule_GetPointer(op, PyExpat_CAPSULE_NAME);
+    if (p != NULL) {
+        PyMem_Free(p);
+    }
```
Comment (vstinner) [inline Modules/pyexpat.c]: The if is not needed: PyMem_Free() accepts NULL. Please remove it.

File: Misc/NEWS.d/next/Core and Builtins/2021-01-02-23-32-55.bpo-41798.yZZWzQ.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Allocate the ``expat_CAPI`` on the heap memory to avoid more module instances
+hold a single ``expat_CAPI`` object after pyexpat ported to multiphase
+initialization (:pep:`489`).
```
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2021-01-02-23-32-55.bpo-41798.yZZWzQ.rst]: I don't think that this change should be mentioned to end users. Only the actual conversion to multiphase init is worth to be mentionned (IMO).

File: Modules/pyexpat.c [hunk 3]
```diff
@@ -1921,40 +1930,42 @@ pyexpat_exec(PyObject *mod)
     MYCONST(XML_PARAM_ENTITY_PARSING_ALWAYS);
 #undef MYCONST
 
-    static struct PyExpat_CAPI capi;
+    struct PyExpat_CAPI *capi = PyMem_Calloc(1, sizeof(struct PyExpat_CAPI));
```
Comment (vstinner) [inline Modules/pyexpat.c:1931]: You need to check for memory allocation failure (check for NULL and call PyErr_NoMemory).

Comment (shihai1991) [reply Modules/pyexpat.c:1931]: Thanks, done.

File: Modules/pyexpat.c (hunk 1 above)
Comment (shihai1991) [reply Modules/pyexpat.c]: Oh, I don't mention this detail. Updated.

File: Misc/NEWS.d/next/Core and Builtins/2021-01-02-23-32-55.bpo-41798.yZZWzQ.rst (hunk 2 above)
Comment (shihai1991) [reply Misc/NEWS.d/next/Core and Builtins/2021-01-02-23-32-55.bpo-41798.yZZWzQ.rst]: Removed.

**Comment by shihai1991** [comment]
Thanks victor for your review and merge.

**Pull request merged**